### ISSUE TITLE
copyq's main window destruction becomes laggy 

### ISSUE THREAD

**Issue by amosbird** [hluk/CopyQ#986]
copyq's main window destruction becomes laggy and somehow breaks the `Return` key behavior (content doesn't get pasted into last window).

https://la.wentropy.com/VtIV.webm

env:

CopyQ Clipboard Manager v3.5.0-25-g9edc672f
Archlinux
i3wm

**Comment by scottkosty** [comment]
@hluk will have a better idea, but I would suggest a git bisect to figure out which commit caused the change in behavior (or whether e.g. it is an updated Qt library that caused it). Usually for an issue like this a git bisect takes much less time to figure out the root problem than other debugging strategies.

Also, please always give the Qt version you compiled against. This can be found with `copyq --version`.

Thanks for the report. I hope we figure out the issue. It certainly sounds like a frustrating issue to deal with.

**Comment by amosbird** [comment]
Hmm, it seems like `copyq toggle` somehow triggers memory leak in i3wm. After restarting i3wm session, the window destruction goes smoothly again.

**Comment by amosbird** [comment]
Is there a way to avoid copyq window recreate? I'm thinking of using i3wm's scratchpad for that.

**Comment by hluk** [comment]
> Is there a way to avoid copyq window recreation?

Not sure about that. I leave the native window handling to Qt except for stealing focus when "show window" on global shortcut is activated.

> I'm thinking of using i3wm's scratchpad for that.

No idea how scratchpad works.

Is there some way I can reproduce the issue? I quickly tested it and it works as expected.

**Comment by amosbird** [comment]
It's a bit hard to get an effective reproduce. Usually it takes a day or two to make the laggy happens.

> No idea how scratchpad works.

I mean, to keep the copyq window always open, such as pressing `Enter` and paste without closing. If it allows a custom script to run (in order to hide the main window), this issue could be workarounded.

**Comment by hluk** [comment]
To keep the main window open, you can try this:

- disable "Always on Top" (General config tab),
- disable "Close When Unfocused" (General config tab),
- disable "Close main window" (History config tab).

Though the behavior will also depend on the window manager.

**Comment by amosbird** [comment]
> Though the behavior will also depend on the window manager.

Yeah, that's why a script hook might be needed to run certain command (hiding the window)
